compiler/util/src/org/jetbrains/kotlin/utils/SmartIdentityTable.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.utils
+import com.intellij.openapi.util.Factory
+import java.util.IdentityHashMap
+
+/**
+ * [SmartIdentityTable] is a Map like structure that uses reference identity for keys.
+ * It uses 2 arrays to store keys & values until the number of entries stored is larger than 10.
```

Comment:
Maybe `[ARRAY_UNTIL_SIZE]` instead of 10

Reply:
done

Reply:
done